### PR TITLE
Add general maintainers role for astropy core package

### PR DESCRIPTION
**Pull request by hamogu**
This has been discussed in the dev telecon 2022-02-03 and there was general agreement to add such a role. This PR is a draft of that role description; unfortunately I don't remember exactly how we described this role in the telecon, so this is based on my notes, my recollection and the discussion in https://github.com/astropy/astropy-project/issues/136

@bsipocz You had a very good description of this role in the last dev telecon, but I was not fast enough to write that down. Could you please have a look at this and comment or suggest changes to include what you said?

x-ref: https://github.com/astropy/astropy-project/issues/136

### CODE REVIEW

File: roles.json [hunk 1]
```diff
@@ -420,6 +437,7 @@
                     "Stuart Littlefair",
                     "Adrian Price-Whelan",
                     "Erik Tollerud",
+                    "Eero Vaher",
```
Comment (pllim) [inline roles.json:440]: Is this supposed to be here? I thought this is #471

**Comment by mwcraig** [comment]
LGTM

File: roles.json [hunk 2]
```diff
@@ -395,7 +395,24 @@
...
+            "unfilled"
+        ],
+        "role-head": "Core astropy package maintainer (general)",
+        "responsibilities": {
+            "description": "Maintain the astropy core package in aspects that are not specific to a single sub-package",
+            "details": [
+                "Evaluating new pull requests for quality, API consistency, Astropy coding standards, and appropriateness within the overall astropy ecosystem, in particular for pull requests spanning multiple sub-packages",
+                "Merging Pull Requests",
+                "Maintain, review, and advocate for useful interaction between multiple sub-packages", 
+                "Keeping track of frequent contributors and their relevant areas of expertise"
```
Comment (eteq) [inline roles.json:410]: In the telecon there was some discussion of the need to distinguish this from the devops role. Looking at the exact wording I do see some overlap here, particularly with the "Merge non-controversial pull requests" aspects of DevOps.

Concretely I think maybe a change like this is warrented:
```suggestion
                "Evaluating new pull requests for quality, API consistency, Astropy coding standards, and appropriateness within the overall astropy ecosystem, in particular for pull requests spanning multiple sub-packages",
                "Merging Pull Requests that are non-controversial or in consultation with relevant subpackage maintainers",
                "Maintain, review, and advocate for useful interaction between multiple sub-packages", 
                "Keeping track of frequent contributors and their relevant areas of expertise"
```

I *also* think there's some question of how much of this work is something the DevOps role does as part of triaging/labeling/etc. That is, in practicle I think the "keeping track" may be more logically connected to the triaging/labeling process. But maybe that really means that most devops folks should *also* be generalists?  Or should we add this last line to the devops role?

**Review by eteq — APPROVED** [review]
Happy with this broadly, but I've suggested what I can remember as part of the discussion in the telecon of the boundary between the generalist and the devops role.

File: roles.json (hunk 2 above)
Comment (eteq) [reply roles.json:410]: Alternative wording:
```suggestion
                "Evaluating new pull requests for quality, API consistency, Astropy coding standards, and appropriateness within the overall astropy ecosystem, in particular for pull requests spanning multiple sub-packages",
                "Merging Pull Requests that are non-controversial or after reaching out to relevant subpackage maintainers",
                "Maintain, review, and advocate for useful interaction between multiple sub-packages", 
                "Keeping track of frequent contributors and their relevant areas of expertise"
```

Comment (hamogu) [reply roles.json:410]: After zoom discussion with @eteq we agreed on this wording. For reference: Our thought is that we want the generalists to merge stuff that's uncontroversial, just like we (Eric and I) want other sub-package maintainers to merge PRs outside their sub-pacakges that are clearly uncontroversial (typos in the docs would be an extreme example,  other PRs are that clearly uncontroversial in the sense that they are discussed on the PR in detail and everyone agrees also fall into this category), but also PRs that go beyond that. The language "after reaching out" is supposed to mean that sub-package maintainers should have a chance to weigh in, but if they don't reply after a reasonable time, or if a sub-package does not have a maintainer, the generalist can merge the PR. Please comment on the wording if you havea better idea on how to say it.

**Comment by hamogu** [comment]
I've added "triaging" to this new role and, for consistency, also to the existing sub-package maintainers roles. I think it's unfair to leave that all to DevOps and there might be cases that that maintainers can actually triage better. a\Adding labels is also non-destructive, so if anyone adds a wrong one, somebody else can fix it any time. The problem is that we don't have enough people helping out (when @pllim is on vacation it's just not happening), not that too many people contribute.

**Pull request merged**